addons/im_livechat/models/im_livechat_channel.py

```diff
@@ -58,14 +65,64 @@ def _default_default_message(self):
     chatbot_script_count = fields.Integer(string='Number of Chatbot', compute='_compute_chatbot_script_count')
     rule_ids = fields.One2many('im_livechat.channel.rule', 'channel_id', 'Rules')
 
+    _concurrent_session_greater_than_zero = models.Constraint(
+        "CHECK(concurrent_session = 'limited' OR concurrent_session_number > 0)",
+        "Concurrent session number should be greater than zero.",
+    )
+
     def _are_you_inside(self):
         for channel in self:
             channel.are_you_inside = self.env.user in channel.user_ids
 
-    @api.depends('user_ids.im_status')
+    @api.depends(
+        "user_ids.im_status",
+        "channel_ids.livechat_active",
+        "channel_ids.last_interest_dt",
+        "channel_ids.rtc_session_ids",
+    )
     def _compute_available_operator_ids(self):
+        session_count_by_operator = {}
+        operator_in_call = self.env["res.partner"]
+        if any(r.concurrent_session == "limited" for r in self):
```

Comment:
```suggestion
        if limited_channels := self.filtered(c: c.concurrent_session == "limited"):
```

Would avoid looping twice through all records